kornia/augmentation/_2d/intensity/channel_dropout.py

```diff
@@ -0,0 +1,88 @@
+from typing import Any, Dict, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.core.check import KORNIA_CHECK_IS_TENSOR, KORNIA_CHECK_SHAPE
+
+
+class RandomChannelDropout(IntensityAugmentationBase2D):
+    r"""Drops channels in the input image.
+
+    In case batch of images is passed, for each image random channels will be dropped.
+
+    .. image:: _static/img/RandomChannelDropout.png
+
+    Args:
+        channel_drop_range: range from which we choose the number of channels to drop.
+        fill_value: pixel value for the dropped channel.
+        same_on_batch: apply the same transformation across the batch.
+        p: probability of applying the transformation.
+        keepdim: whether to keep the output shape the same as input ``True`` or broadcast it
+          to the batch form ``False``.
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> img = torch.arange(1*2*2*2.).view(1,2,2,2)
+        >>> RandomChannelDropout()(img)
+        tensor([[[[0., 1.],
+                  [2., 3.]],
+        <BLANKLINE>
+                 [[0., 0.],
+                  [0., 0.]]]])
+    """
+
+    def __init__(
+        self,
+        channel_drop_range: Tuple[int, int] = (1, 1),
+        fill_value: Union[int, float] = 0,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim)
+        self.min_channels = channel_drop_range[0]
+        self.max_channels = channel_drop_range[1]
+        if not 1 <= self.min_channels <= self.max_channels:
+            raise ValueError(f"Invalid channel_drop_range. Got: {channel_drop_range}")
+
+        self.fill_value = fill_value
+
+    def generate_parameters(self, shape: torch.Size) -> Dict[str, Tensor]:
+        B, C, _, _ = shape
+        if C <= 1:
+            raise ValueError("Channel dropout can be applied only to multichannel images")
+
+        if self.max_channels >= C:
+            raise ValueError("Cannot drop all channels in ChannelDropout.")
+
+        # [B] - for each batch drop random amount of channels
+        num_drop_channels = torch.randint(low=self.min_channels, high=self.max_channels + 1, size=(B,))
+
+        # B, C
+        channels_to_drop = torch.argsort(torch.rand(B, C))
```

Comment:
same for the samplers

Reply:
Not sure whether it is possible to rewrite this torch.argsort(torch.rand(B, C)) nicely via Sample API, ChannelShuffle does this in the same way

Reply:
I feel like you may have a custom `RandomGenerator`. For example,

https://github.com/kornia/kornia/blob/master/kornia/augmentation/random_generator/_2d/gaussian_blur.py